Plugins/SQL/Targets/SQLite.cpp

```diff
@@ -0,0 +1,218 @@
+#if defined(NWNX_SQL_SQLITE_SUPPORT)
+
+#include "SQLite.hpp"
+#include "Services/Config/Config.hpp"
+#include "API/Globals.hpp"
+#include "API/CExoBase.hpp"
+#include "Platform/FileSystem.hpp"
+
+#include <string.h>
+#include <sqlite3.h>
+
+namespace SQL {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+using namespace Platform::FileSystem;
+
+SQLite::SQLite()
+{
+    m_dbName = "database";
+    m_stmt = nullptr;
+    m_lastError = "";
+    m_paramCount = 0;
+}
+
+SQLite::~SQLite()
+{
+    sqlite3_close(m_dbConn);
+}
+
+void SQLite::Connect(NWNXLib::ViewPtr<NWNXLib::Services::ConfigProxy> config)
+{ 
+    if (auto database = config->Get<std::string>("DATABASE"))
+    {
+        m_dbName = (*database).c_str();
```

Comment:
FYI, `(*var).something` is equivalent to `var->something`. Often people consider the second easier to read.